core/src/main/java/org/infinispan/configuration/cache/MemoryConfigurationBuilder.java

```diff
@@ -336,9 +336,6 @@ public MemoryConfigurationBuilder read(MemoryConfiguration template) {
       if (attributes.attribute(MemoryConfiguration.MAX_COUNT).isModified()) {
          updateMaxCount(maxCount());
       }
-      if (attributes.attribute(MemoryConfiguration.MAX_SIZE).isModified()) {
```

Comment:
I have to admit just from looking at the diff this looks wrong. Are we able to just remove the MemoryStorageConfiguration instead? If this is indeed correct, I believe we need to update the listener registered in the constructor as well right?